include/itkAttenuationImageFilter.h

```diff
@@ -0,0 +1,204 @@
+  itkSetMacro(Direction, unsigned int);
+
+  itkGetConstMacro(SamplingFrequencyMHz, float);
+  itkSetMacro(SamplingFrequencyMHz, float);
+
+  itkGetConstMacro(FrequencyBandStartMHz, float);
+  itkSetMacro(FrequencyBandStartMHz, float);
+
+  itkGetConstMacro(FrequencyBandEndMHz, float);
+  itkSetMacro(FrequencyBandEndMHz, float);
+
+  itkGetConstMacro(ConsiderNegativeAttenuations, bool);
+  itkSetMacro(ConsiderNegativeAttenuations, bool);
+
+  itkGetConstMacro(PadPixelUpperBounds, unsigned int);
+  itkSetMacro(PadPixelUpperBounds, unsigned int);
+  itkGetConstMacro(PadPixelLowerBounds, unsigned int);
+  itkSetMacro(PadPixelLowerBounds, unsigned int);
+
+  itkGetConstMacro(PadSpatialUpperBoundsMM, float);
+  itkSetMacro(PadSpatialUpperBoundsMM, float);
+  itkGetConstMacro(PadSpatialLowerBoundsMM, float);
+  itkSetMacro(PadSpatialLowerBoundsMM, float);
+
+  // Alias for setting direction of RF waveform in data collection
+  void
+  SetScanDirection(unsigned int direction)
+  {
+    this->SetDirection(direction);
+  };
+  unsigned int
+  GetScanDirection()
+  {
+    return this->GetDirection();
+  };
+
+  void
+  PrintSelf(std::ostream & os, Indent indent) const override;
+
+protected:
+  AttenuationImageFilter();
+  ~AttenuationImageFilter() override = default;
+
+  /** Initialize some accumulators before any chunks are processes */
+  void
+  BeforeThreadedGenerateData() override;
+
+  /** Do final mean and variance computation from data accumulated in threads.
+   */
+  void
+  AfterThreadedGenerateData() override;
+
+  void
+  DynamicThreadedGenerateData(const OutputRegionType &) override;
+
+  const ImageRegionSplitterBase *
+  GetImageRegionSplitter() const override;
+
+  /** Compute attenuation between two pixels in the RF spectra vector image.
+   *  Assumes that image spacing is in MM. */
+  OutputPixelType
+  ComputeAttenuation(const InputIndexType & end, const InputIndexType & start) const;
+
+private:
+  /** Label value indicating pixel should be included in analysis */
+  MaskPixelType m_LabelValue = 1;
+
+  /** RF scanline direction */
+  unsigned int m_Direction = 0;
+
+  /** RF sampling frequency */
+  float m_SamplingFrequencyMHz = 0.0f;
+
+  /** RF frequency band */
+  float m_FrequencyBandStartMHz = 0.0f;
+  float m_FrequencyBandEndMHz = 0.0f;
+
+  /** Optionally discard negative attenuation estimates
+   *  so that they are not considered in statistic computations.
+   *  Negative attenuation implies that a signal strengthened
+   *  while passing through tissue and may result from
+   *  sampling error or external interference. */
+  bool m_ConsiderNegativeAttenuations = false;
+
+  /** Skip attenuation estimation for a fixed number
+   *  of pixels at the start of an inclusion region.
+   *  Applied before spatial padding.
+   *  Can help with uncertainty at borders of mask.
+   */
+  unsigned int m_PadPixelUpperBounds = 0;
+  /** Skip attenuation estimation at the end of an inclusion region. */
```

Comment:
Are lower and upper bounds docstrings mixed up?

Reply:
Yes, after investigating it looks like I got these mixed up. Preparing a fix.